Python/pywarpx/picmi.py

```diff
@@ -163,14 +163,13 @@ def initialize_inputs(self, layout,
             self.species.rigid_advance = 1
             self.species.zinject_plane = injection_plane_position
 
-        for interaction in self.interactions:
-            assert interaction[0] == 'ionization'
-            assert interaction[1] == 'ADK', 'WarpX only has ADK ionization model implemented'
-            self.species.do_field_ionization=1
-            self.species.physical_element=self.particle_type
-            self.species.ionization_product_species = interaction[2].name
-            self.species.ionization_initial_level = self.charge_state
-            self.species.charge = 'q_e'
+    def activate_ionization(self, product_species):
```

Comment:
Thanks for the PR. This looks ok, but is different than what I would have done. I would have put the code here in the  `initialize_inputs` method in the `FieldIonization` class. This more closely follows the pattern of how things are done in PICMI. That class has the references to the species and so can set these parameters that are related to the field ionization setup. Also, there is no need to use `add_new_attr` - these parameters can be set directly, `self.species.do_field_ionization = 1`. `add_new_attr` is there for when the attributes have special characters or is constructed.